nucliadb/src/nucliadb/ingest/fields/base.py

```diff
@@ -596,3 +596,101 @@ async def set_value(self, value: Any):
 
     async def get_value(self) -> Any:
         raise NotImplementedError()
+
+
+class ExtractedTextIndex:
+    def __init__(self, et: ExtractedText):
+        self.et = et
+        self.step_size = 1024
+        self.index: dict[int, int] = self.create_index()
+
+    def create_index(self) -> dict[int, int]:
+        # We assume that the dictionary will preserve the order of insertion
+        # (Python 3.7+), so we can use it to create an index
+        # that maps the position of the text in the ExtractedText to the byte
+        # position in the serialized ExtractedText.
+        index: dict[int, int] = {}
+        encoded = self.et.SerializeToString()
+        # Initialize the index with the start position by searching for the first two characters.
+        # This is a trick to know where the protobuff header ends.
+        text_starts_at = encoded.index(self.et.text[:2].encode())
+        index[0] = text_starts_at
+
+        for i in range(self.step_size, len(self.et.text), self.step_size):
+            text_slice = self.et.text[0:i]
+            index[i] = len(text_slice.encode()) + text_starts_at
```

Comment:
Here you are encoding the whole text up to this point. It would be better if you only encoded the current chunk (`[i-step_size : i]`) and then added the position of the previous chunk to the position of this one. This way, you only encoded each chunk once (right now you are encoding each part of the text chunks² times on average)